infra/modules/vpc/main.tf

```diff
@@ -0,0 +1,33 @@
+# This implementation of the standard terraform VPC module
+# provides a few variations of a basic VPC, based on input
+# variables
+
+module "vpc" {
```

Comment:
nit, I've been trying to reduce language specific to cloud providers and tech stacks, and focus our naming based on the function or purpose of the things. Azure calls it VNet or Virtual Network, while GCP and AWS call them Virtual Private Cloud, but ultimately it's a virtualized network, so I think we should just call this module `"network"` rather than `"vpc"`. I think that's also helpful for non-infra engineers who aren't as familiar with the industry jargon but are still familiar with the fundamental systems concepts.

So I'd suggest we do a find/replace `vpc` -> `network` (e.g. `vpc_name` -> `network_name`, etc)

i've adopted this approach for other parts of the codebase too:
- "container-image-repository" rather than "elastic container registry"

If in the future we want to support multiple cloud providers we can have consistent naming schemes for them so that engineers can move between projects and still know their way around the codebase (e.g. it'll be easier to translate between `network-aws` and `network-azure` than it would be to do translate between `vpc` and `vnet`

Reply:
I didn't realize we were going for cloud-agnostic and I can see where in certain places (e.g. instead of calling the folder infra/vpcs calling it infra/networks.  but this module is providing a aws vpc and I doubt when you get to Azure you will want to pour azure stuff into this module with a ton of conditional stuff.  rather, i'd counter-propose that we call this infra/modules/aws-vpc/main.tf  does that make sense?

Reply:
or did you answer my issue with the suggestion of "network-aws" that would do the trick

Reply:
as I start to implement this, i just want to clarify....  there are parameters being passed back and forth (like vpc_id and vpc_name,  these are things that map to the AWS resource definition in terraform.  We will be obfuscating things if we drive the generic names to deeply into these modules, so I drew the line there.